### ISSUE TITLE
Platform compatibility analyzer doesn't consider attributes on local functions

### ISSUE THREAD

**Issue by terrajobst** [dotnet/roslyn-analyzers#5975]
### Analyzer

**Diagnostic ID**: CA1416

### Analyzer source

**SDK**: Version: 7.0.0-preview.3.22175.4

### Describe the bug

It looks like the platform compat analyzer doesn't consider platform attributes
on local functions.

### Steps To Reproduce

The following code doesn't produce any diagnostics:

```C#
static class Program
{
    void Main()
    {
        [SupportedOSPlatform("ios")]
        void Test()
        {
            var x = Process.GetProcesses();
        }
    }
}
```

However, making `Test` a regular method will produce a diagnostic:

```C#
static class Program
{
    void Main()
    {
    }

    [SupportedOSPlatform("ios")]
    void Test()
    {
        var x = Process.GetProcesses();
    }

}
```

> CA1416: This call site is reachable on: 'ios', 'maccatalyst'. 'Process.GetProcesses()' is unsupported on: 'ios'.

### Expected behavior

Platform support attributes on local functions should be considered just like any other method.

### Actual behavior

Platform support attributes on local functions are entirely ignored.
